packages/x-components/src/x-modules/url/store/__tests__/getters.spec.ts

```diff
@@ -26,23 +20,51 @@ describe('testing url module getters', () => {
   it('re-maps values using the config', () => {
     expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
       query: 'salmorejo',
-      relatedTag: ['with eggs'],
+      tag: ['with eggs'],
       sort: 'default',
-      warehouse: 12345,
-      isLoadedFromUrl: false
+      warehouse: 12345
+    });
+  });
+
+  it('removes all the parameters from the url that are the default values in the state', () => {
+    resetUrlStateWith(store, {
+      params: {
+        page: 2,
+        query: 'doramion',
+        scroll: 444
+      }
+    });
+
+    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
+      query: 'doramion',
+      page: 2,
+      scroll: 444
+    });
+
+    resetUrlStateWith(store, { params: { page: 1, query: 'lego', scroll: 0 } });
+
+    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
+      query: 'lego'
     });
   });
 
-  it('maps the param key from the state with the config one or itself', () => {
-    expect(store.getters.urlMappedParamNames).toEqual<UrlGetters['urlMappedParamNames']>({
-      query: 'q',
-      filters: 'filters',
-      relatedTag: 'tag',
-      page: 'page',
-      scroll: 'scroll',
-      sort: 'sort',
-      warehouse: 'warehouse',
-      store: 'store'
+  it('removes all the empty extra params', () => {
+    resetUrlStateWith(store, {
+      params: {
+        query: 'doramion',
+        warehouse: '11111'
+      }
+    });
+
+    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
+      query: 'doramion',
+      warehouse: '11111'
+    });
+
+    resetUrlStateWith(store, { params: { warehouse: '' } });
+
+    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
```

Comment:
The test is failling because the getter is returning the warehouse: "". In the getter, the warehouse is inside the params, so you are not cleaning the extra param when they are empty: 

![image](https://user-images.githubusercontent.com/6597815/137310210-fa121ed7-a8c1-41ca-8f73-d42e33d5a2d2.png)

Reply:
` it('removes all the empty extra params', () => {
    resetUrlStateWith(store, {
      params: {
        query: 'doramion'
      },
      extraParams: {
        warehouse: '11111'
      }
    });

    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
      query: 'doramion',
      warehouse: '11111'
    });

    resetUrlStateWith(store, {
      params: {
        query: 'doramion'
      },
      extraParams: { warehouse: '' }
    });

    expect(store.getters.urlParams).toEqual<Partial<UrlGetters['urlParams']>>({
      query: 'doramion'
    });`